openedx/platform-plugin-aspects
# feat: migrates SupersetXBlock

### Description

Migrates the SupersetXBlock code from https://github.com/eduNEXT/platform-plugin-superset

and

* use settings.SUPERSET_CONFIG instead of storing sensitive data in XBlock fields (superset_url, superset_username, superset_password)
* modifies settings.SUPERSET_CONFIG to better distinguish between the (public) service_url and the (docker) `internal_service_url`
* simplifies the XBlock by using StudioEditableXBlockMixin
* removes unused code and files.

Also adds a "Go to Superset" link to the Instructor Dashboard, to allow users to navigate directly to Superset.

### Testing instructions

See https://github.com/openedx/tutor-contrib-aspects/pull/642  for manual testing instructions.


### Screenshots

XBlock edit screen:

![image](https://github.com/openedx/platform-plugin-aspects/assets/7556571/c3323529-1857-45d5-93e1-baf89ef7f5f9)

XBlock staff/instructor view:

![image](https://github.com/openedx/platform-plugin-aspects/assets/7556571/fa94dcbe-1bbc-4ef6-85d2-e8da645cd715)

XBlock learner view:

![image](https://github.com/openedx/platform-plugin-aspects/assets/7556571/55ba6f67-21f3-4e8b-864c-13c8361374be)

Instructor Dashboard "Go to Superset" link:

![image](https://github.com/openedx/platform-plugin-aspects/assets/7556571/e956fb10-98c9-4fda-a928-5b948908e206)


**Merge checklist:**
Check off if complete *or* not applicable:
